### PR TITLE
feat(cli): add code-mod utility in CLI that refactors old ":import" to the new "@st-import"

### PR DESCRIPTION
**Pull request by barak007**
expose cli that can apply code mods to stylable files

only support `st-import-to-at-import` for now.

### CODE REVIEW

File: packages/cli/src/code-mods/code-mods.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        const apply = registeredMods[id];
+        if (!apply) {
+            log(`Unknown mod ${id}`);
+        } else {
+            loadedMods.add({ id, apply });
+        }
+    }
+
+    if (loadedMods.size !== mods.length) {
+        return;
```
Comment (tomrav) [inline packages/cli/src/code-mods/code-mods.ts:41]: missing log

File: packages/cli/src/code-mods/st-import-to-at-import.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        for (const [as, name] of named) {
+            if (as === name) {
+                namedParts.push(name);
+            } else {
+                namedParts.push(`${name} as ${as}`);
+            }
+        }
+        const keyFramesParts: string[] = [];
+        for (const [as, name] of keyframes) {
+            if (as === name) {
```
Comment (tomrav) [inline packages/cli/src/code-mods/st-import-to-at-import.ts:40]: extract to function?

File: packages/cli/src/code-mods/st-import-to-at-import.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                params += ', ';
+            }
+            params += `keyframes(${keyFramesParts.join(', ')})`;
+        }
+        params += ']';
+    }
+
+    params += ` from ${JSON.stringify(importObj.request)};`;
+
+    return postcss.atRule({ name: 'st-import', params });
```
Comment (tomrav) [inline packages/cli/src/code-mods/st-import-to-at-import.ts:59]: This function raises some concerns about changing order and dropping comments/spacing

File: packages/cli/test/cli-codemod.spec.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    let tempDir: ITempDirectory;
+
+    beforeEach(async () => {
+        tempDir = await createTempDirectory();
+    });
+    afterEach(async () => {
+        await tempDir.remove();
+    });
+
+    describe('Code Mods', () => {
```
Comment (tomrav) [inline packages/cli/test/cli-codemod.spec.ts:21]: change the top level describe to Codemode CLI and drop this describe

File: packages/cli/test/cli-codemod.spec.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                '--mods',
+                'st-import-to-at-import',
+            ]);
+
+            expect(stderr).equal('');
+
+            console.log(stdout);
+
+            const dirContent = loadDirSync(tempDir.path);
+            expect(dirContent['style.st.css']).equal('@st-import Name from "./x";');
```
Comment (tomrav) [inline packages/cli/test/cli-codemod.spec.ts:40]: Missing tests to validate named imports

**Review by tomrav — CHANGES_REQUESTED** [review]
Nice feature addition to our CLI offerings, missing documentation (in the readme?)

File: packages/cli/src/code-mods/code-mods.ts (hunk 1 above)
Comment (barak007) [reply packages/cli/src/code-mods/code-mods.ts:41]: no it's logging before that

**Comment by PeterShershov** [comment]
Great addition, we need this! 
Please merge and deploy ASAP 😄  @barak007 @tomrav

**Comment by tomrav** [comment]
this currently creates double semicolons after the from path, need to fix.

**Comment by tzachbon** [comment]
It should be mentioned in `stylable.io`.

File: packages/cli/src/code-mods/st-import-to-at-import.ts [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { parsePseudoImport, Diagnostics, Imported } from '@stylable/core';
+import postcss, { Root, AtRule } from 'postcss';
+
+export function stImportToAtImport(ast: Root, messages: string[]) {
+    ast.walkRules((rule) => {
+        if (rule.selector === ':import') {
+            const diagnostics = new Diagnostics();
+            const importObj = parsePseudoImport(rule, '*', diagnostics);
+            if (diagnostics.reports.length) {
```
Comment (barak007) [inline packages/cli/src/code-mods/st-import-to-at-import.ts:9]: Handle info

File: packages/cli/src/code-mods/st-import-to-at-import.ts (hunk 3 above)
Comment (barak007) [reply packages/cli/src/code-mods/st-import-to-at-import.ts:47]: True we don't want this to be a blocker so we will handle it in a later time

**Pull request merged**